core/src/main/scala/org/apache/spark/SSLOptions.scala

```diff
@@ -231,5 +230,45 @@ private[spark] object SSLOptions extends Logging {
       enabledAlgorithms)
   }
 
+  private def getPassword(
+      conf: SparkConf,
+      hadoopConf: Configuration,
+      ns: String,
+      parameter: String,
+      default: Option[String]): Option[String] = {
+    var parameterValue = conf.getWithSubstitution(s"$ns.$parameter")
+      .orElse(Option(hadoopConf.getPassword(s"$ns.$parameter")).map(new String(_)))
+      .orElse(default)
+    if (parameterValue.isDefined && default.isDefined && parameterValue.get != default.get) {
```

Comment:
I have no idea so I'm asking, Is there a case which default can be `None`? If then I guess we also need to warn if `(parameterValue.isDefined && !default.isDefined)` is true. 

One liner (but long) would be `if (parameterValue.isDefined && (!default.isDefined || (parameterValue.get != default.get)))`

Reply:
It's a valid possibility so changed.